entr.c

```diff
@@ -160,9 +160,12 @@ main(int argc, char *argv[]) {
 	setenv("PAGER", "/bin/cat", 0);
 
 	/* ensure a shell is available to use */
-	setenv("SHELL", "/bin/sh", 0);
+	if (0 != setenv("SHELL", "/bin/sh", 0))
+		err(1, "cannot set SHELL environment variable to /bin/sh");
```

Comment:
This should probably be written 
```
if (setenv("SHELL", "/bin/sh", 0) == -1)
```

Under what condition would this fail? I don't see how to reproduce this.